internal/policy/verify.go

```diff
@@ -530,105 +502,83 @@ func verifyEntry(ctx context.Context, repo *git.Repository, policy *State, attes
 	return nil
 }
 
-func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, entry *rsl.ReferenceEntry) error {
-	// 1. Find authorized public keys for tag's RSL entry
-	trustedKeys, err := policy.FindPublicKeysForPath(ctx, fmt.Sprintf("git:%s", entry.RefName))
+func verifyTagEntry(ctx context.Context, repo *gitinterface.Repository, policy *State, attestationsState *attestations.Attestations, entry *rsl.ReferenceEntry) error {
+	// Find authorized public keys for tag's RSL entry
+	verifiers, err := policy.FindVerifiersForPath(fmt.Sprintf("%s:%s", gitReferenceRuleScheme, entry.RefName))
 	if err != nil {
 		return err
 	}
 
-	if len(trustedKeys) == 0 {
-		allKeys, err := policy.PublicKeys()
+	if len(verifiers) == 0 {
+		return nil
+	}
+
+	var authorizationAttestation *sslibdsse.Envelope
+	if attestationsState != nil {
+		authorizationAttestation, err = getAuthorizationAttestation(repo, attestationsState, entry)
 		if err != nil {
 			return err
 		}
-
-		// FIXME: decide if we want to pass around map or slice for these APIs
-		for _, key := range allKeys {
-			trustedKeys = append(trustedKeys, key)
-		}
 	}
 
-	// 2. Find commit object for the RSL entry
-	commitObj, err := gitinterface.GetCommit(repo, entry.ID)
-	if err != nil {
-		return err
-	}
-
-	// 3. Use each trusted key to verify signature
+	// Use each verifier to verify signature
 	rslEntryVerified := false
-	for _, key := range trustedKeys {
-		err := gitinterface.VerifyCommitSignature(ctx, commitObj, key)
+	for _, verifier := range verifiers {
+		err := verifier.Verify(ctx, entry.ID, authorizationAttestation)
 		if err == nil {
 			// Signature verification succeeded
 			rslEntryVerified = true
 			break
-		}
-		if errors.Is(err, gitinterface.ErrUnknownSigningMethod) {
-			// We encounter this for key types that can be used for gittuf
-			// policy metadata but not Git objects
-			continue
-		}
-		if !errors.Is(err, gitinterface.ErrIncorrectVerificationKey) {
+		} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
 			// Unexpected error
 			return err
 		}
-		// Haven't found a valid key, continue with next key
+		// Haven't found a valid verifier, continue with next key
 	}
```

Comment:
No action needed for this PR, but something to think about: the entry verify flow is repeated in very similar ways across the different verify funcs. We should see if there are ways to consolidate these so they don't diverge in subtle ways.

Reply:
Yes agreed, we consolidated some of this in verifier.Verify but we could probably do a consolidation of looping over verifiers as well.

Reply:
#454